lib/logstash/plugin_mixins/validator_support/field_reference_validation_adapter.rb

```diff
@@ -16,10 +16,12 @@ module ValidatorSupport
 
       FieldReferenceValidationAdapter = NamedValidationAdapter.new(:field_reference) do |value|
         break ValidationResult.failure("Expected exactly one field reference, got `#{value.inspect}`") unless value.kind_of?(Array) && value.size <= 1
-        break ValidationResult.success(nil) if value.empty? || value.first.nil?
 
         candidate = value.first
 
+        break ValidationResult.success(nil) if value.empty? || candidate.nil?
+        break ValidationResult.success(candidate) if candidate.empty? # compatibility with LS >= 7.x field_reference validation logic
```

Comment:
🤔 

The implementation we are attempting to approximate is indeed different, but it returns `nil` if the candidate is empty.

> ~~~ ruby
>             return [true, nil] if value.empty? || value.first.nil? || value.first.empty?
> ~~~
> [logstash-core/lib/logstash/config/mixin.rb:618@v7.11.0](https://github.com/elastic/logstash/blob/v7.11.0/logstash-core/lib/logstash/config/mixin.rb#L618)

```suggestion
        break ValidationResult.success(nil) if value.empty? || candidate.nil? || candidate.empty?
```

Reply:
thanks, I should have checked the existing implementation. 
blindly assumed the field reader will differentiate a `''` vs `nil` when the option isn't set.
updated and confirmed behavior with an actual (imap input) plugin.